game.cpp

```diff
@@ -1066,126 +1200,150 @@ void test_game() //!OCLINT tests may be many
 #endif
 
 #ifdef FIX_ISSUE_254
+    {
+        game g;
+        put_player_on_food(g.get_player(0), g.get_food()[0]);
+        g.tick();
+        assert(is_eaten(g.get_food()[0]))
+    }
+#endif
+
+//#define FIX_ISSUE_340
+#ifdef FIX_ISSUE_340
+// make sure that eat_nth_food() throws a logic_error when the food is already eaten
   {
-    game g;
-    put_player_on_food(g.get_player(0), g.get_food()[0]);
-    g.tick();
-    assert(is_eaten(g.get_food()[0]))
+    try {
+      game g; //by default one uneaten food
+      assert(has_food(g));
+      eat_nth_food(g,0);
+      assert(!has_food(g));
+      eat_nth_food(g,0); // throws exception
+    }
+    catch ( const std::exception& e ) {
+      assert(e == std::logic_error); // this is still wrong: make sure the error caught is the same as the error thrown
+    }
```

Comment:
I don't know what's our policy on try-catch statements. I feel this should be the role of an assert statement? I would need a second opinion on this - @richelbilderbeek ?